sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/ClientMetricsTest.java

```diff
@@ -0,0 +1,524 @@
+
+public class ClientMetricsTest extends BatchTestBase {
+
+    private CosmosClient client;
+    private CosmosContainer container;
+    private String databaseId;
+    private String containerId;
+    private MeterRegistry meterRegistry;
+
+    @Factory(dataProvider = "clientBuildersWithDirectSession")
+    public ClientMetricsTest(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+    public void beforeTest() {
+        assertThat(this.client).isNull();
+        assertThat(this.meterRegistry).isNull();
+
+        this.meterRegistry = ConsoleLoggingRegistryFactory.create(1);
+        this.client = getClientBuilder()
+            .clientTelemetryConfig().clientMetrics(this.meterRegistry)
+            .buildClient();
+
+        if (databaseId == null) {
+            CosmosAsyncContainer asyncContainer = getSharedMultiPartitionCosmosContainer(this.client.asyncClient());
+            this.databaseId = asyncContainer.getDatabase().getId();
+            this.containerId = asyncContainer.getId();
+        }
+
+        container = client.getDatabase(databaseId).getContainer(containerId);
+    }
+
+    public void afterTest() {
+        this.container = null;
+        CosmosClient clientSnapshot = this.client;
+        if (clientSnapshot != null) {
+            this.client.close();
+        }
+        this.client = null;
+
+        MeterRegistry meterRegistrySnapshot = this.meterRegistry;
+        if (meterRegistrySnapshot != null) {
+            meterRegistrySnapshot.clear();
+            meterRegistrySnapshot.close();
+        }
+        this.meterRegistry = null;
+    }
+
+    @Test(groups = { "simple" }, timeOut = TIMEOUT, singleThreaded = true)
+    public void createItem() throws Exception {
+        this.beforeTest();
+        try {
+            InternalObjectNode properties = getDocumentDefinition(UUID.randomUUID().toString());
+            CosmosItemResponse<InternalObjectNode> itemResponse = container.createItem(properties);
+            assertThat(itemResponse.getRequestCharge()).isGreaterThan(0);
+            validateItemResponse(properties, itemResponse);
+
+            properties = getDocumentDefinition(UUID.randomUUID().toString());
+            CosmosItemResponse<InternalObjectNode> itemResponse1 = container.createItem(properties, new CosmosItemRequestOptions());
+            validateItemResponse(properties, itemResponse1);
+
+            this.validateMetrics(
+                Tag.of(TagName.OperationStatusCode.toString(), "201"),
+                Tag.of(TagName.RequestStatusCode.toString(), "201/0")
+            );
+
+            this.validateMetrics(
+                Tag.of(
+                    TagName.Operation.toString(), "Document/Create"),
+                Tag.of(TagName.RequestOperationType.toString(), "Document/Create")
+            );
+        } finally {
+            this.afterTest();
+        }
+    }
+
+    @Test(groups = { "simple" }, timeOut = TIMEOUT, singleThreaded = true)
```

Comment:
why we need singleThreaded = true here?

Reply:
Don't think it is needed - explored this when debugging issues with the tests that only showed up in DevOps not locally. Removed it for now.